### PR TITLE
test(bigtable): increase ClientOptions test coverage

### PR DESCRIPTION
**Pull request by dbolduc**
Add missing unit tests for some of the `ClientOptions`. This will give us more confidence to transition the `ClientOptions` implementation to use `Options`

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6945)

### CODE REVIEW

File: google/cloud/bigtable/client_options_test.cc [hunk 1]
```diff
@@ -387,6 +388,81 @@ TEST(ClientOptionsTest, RefreshPeriod) {
...
+}
+
+TEST(ClientOptionsTest, CustomBackgroundThreads) {
+  CompletionQueue cq;
+  auto options = ClientOptions().DisableBackgroundThreads(cq);
+  auto background = options.background_threads_factory()();
+
+  using ms = std::chrono::milliseconds;
+
+  // Schedule some work, it cannot execute because there is no thread attached.
```
Comment (devbww) [inline google/cloud/bigtable/client_options_test.cc:451]: `s/, it/that/` perhaps?

`s/attached/draining the completion queue/` perhaps?

File: google/cloud/bigtable/client_options_test.cc [hunk 2]
```diff
@@ -387,6 +388,81 @@ TEST(ClientOptionsTest, RefreshPeriod) {
...
+  promise<std::thread::id> p;
+  auto background_thread_id = p.get_future();
+  background->cq().RunAsync(
+      [&p](CompletionQueue&) { p.set_value(std::this_thread::get_id()); });
+  EXPECT_NE(std::future_status::ready, background_thread_id.wait_for(ms(1)));
```
Comment (devbww) [inline google/cloud/bigtable/client_options_test.cc:456]: Perhaps 10ms might provide clearer evidence that the "not ready" is not just a scheduling issue.

File: google/cloud/bigtable/client_options_test.cc [hunk 3]
```diff
@@ -387,6 +388,81 @@ TEST(ClientOptionsTest, RefreshPeriod) {
...
+
+  // Verify we can create our own threads to drain the completion queue.
+  std::thread t([&cq] { cq.Run(); });
```
Comment (devbww) [inline google/cloud/bigtable/client_options_test.cc:459]: `#include <thread>`

**Review by dbolduc — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6945)* status: 0 of 1 files reviewed, 6 unresolved discussions (waiting on @devbww)

___
*[google/cloud/bigtable/client_options_test.cc, line 410 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6945#-Mdys20T6cdSDUwTy9Ct-r1-410:-Me08HPq4rxVaXFVDgxC:b-piys02) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/52b30680d3c92bdf9d42b582320a917989c580fe/google/cloud/bigtable/client_options_test.cc#L410)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

?
</blockquote></details>

Maybe "?!?!" is more accurate...... Removed.
___
*[google/cloud/bigtable/client_options_test.cc, line 415 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6945#-Mdys20T6cdSDUwTy9Ct-r1-415:-Me09e2k6qMYYwmlVCGy:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/52b30680d3c92bdf9d42b582320a917989c580fe/google/cloud/bigtable/client_options_test.cc#L415)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

I think we came to a group decision sometime shortly before your tenure to prefer an explicit `absl::nullopt` in such circumstances.  While we didn't go back an change existing examples, I believe we agreed to be explicit in the future.
</blockquote></details>

Done.
___
*[google/cloud/bigtable/client_options_test.cc, line 429 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6945#-Mdys20T6cdSDUwTy9Ct-r1-429:-Me0CqCzCVvstQG3sY5y:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/52b30680d3c92bdf9d42b582320a917989c580fe/google/cloud/bigtable/client_options_test.cc#L429)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

I'm not sure what this is trying to tell me.
</blockquote></details>

Done.
___
*[google/cloud/bigtable/client_options_test.cc, line 451 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6945#-Mdys20T6cdSDUwTy9Ct-r1-451:-Me0DZHI6z9ux5GwNc9i:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/52b30680d3c92bdf9d42b582320a917989c580fe/google/cloud/bigtable/client_options_test.cc#L451)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

`s/, it/that/` perhaps?

`s/attached/draining the completion queue/` perhaps?
</blockquote></details>

Done.
___
*[google/cloud/bigtable/client_options_test.cc, line 456 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6945#-Mdys20T6cdSDUwTy9Ct-r1-456:-Me0D06w-l8C1OQFVV-n:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/52b30680d3c92bdf9d42b582320a917989c580fe/google/cloud/bigtable/client_options_test.cc#L456)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Perhaps 10ms might provide clearer evidence that the "not ready" is not just a scheduling issue.
</blockquote></details>

Done.
___
*[google/cloud/bigtable/client_options_test.cc, line 459 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6945#-Mdys20T6cdSDUwTy9Ct-r1-459:-Me0DA9n5xiZ7gY089Uo:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/52b30680d3c92bdf9d42b582320a917989c580fe/google/cloud/bigtable/client_options_test.cc#L459)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

`#include <thread>`
</blockquote></details>

Done.

**Review by devbww — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6945)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @dbolduc)

___
*[google/cloud/bigtable/client_options_test.cc, line 429 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6945#-Mdys20T6cdSDUwTy9Ct-r1-429:-Me0pnwv4dOwWW01DKwP:b-z77qwj) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/52b30680d3c92bdf9d42b582320a917989c580fe/google/cloud/bigtable/client_options_test.cc#L429)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Sorry ... I missed the "`GE`".

**Pull request merged**